testing/run_tests.py

```diff
@@ -1026,14 +1045,24 @@ def run_impeller_golden_tests(build_dir: str):
                                                          ).joinpath('golden_tests_harvester')
   with tempfile.TemporaryDirectory(prefix='impeller_golden') as temp_dir:
     run_cmd([tests_path, '--working_dir=%s' % temp_dir], cwd=build_dir)
-    num_generated_files = len(os.listdir(temp_dir))
-    if num_generated_files != _NUM_EXPECTED_GENERATED_IMPELLER_GOLDEN_FILES:
-      raise Exception(
-          '`impeller_golden_tests` was expected to generate '
-          f'{_NUM_EXPECTED_GENERATED_IMPELLER_GOLDEN_FILES} files, '
-          f'{num_generated_files} were generated. If this is expected, update '
-          '_NUM_EXPECTED_GENERATED_IMPELLER_GOLDEN_FILES.'
+    with tempfile.NamedTemporaryFile(mode="w",
+                                     prefix="impeller_golden_tests_output") as dir_listing_file:
+      dir_listing = generate_dir_listing(temp_dir)
+      dir_listing_file.write(dir_listing)
+      golden_path = os.path.join("testing", "impeller_golden_tests_output.txt")
+      diff_result = subprocess.run(
+          f'git diff -p {golden_path} {dir_listing_file.name}',
+          check=False,
+          shell=True,
+          stdout=subprocess.PIPE,
+          cwd=os.path.join(BUILDROOT_DIR, "flutter")
       )
+      if diff_result.returncode != 0:
+        print_divider('<')
+        print(f'Unexpected diff in {golden_path}, use `git apply` with the following patch.')
+        print('')
+        print(redirect_patch(diff_result.stdout.decode()))
```

Comment:
Could the diff also be written to a file? That way we might also be able to teach CI to harvest the file, which could provide a better workflow than copy-pasting from the CI logs.

Reply:
I like the idea.  I don't think there is prior art for that though.  Copying and pasting the patch is how the license checker and the formatter work.  You are thinking it could print out an url to a patch file?  So you could do something like `git apply | curl <url>`?

Reply:
I mean `git apply | curl <url>` has the same workflow as copy and pasting the patch.  It's just a smaller copy and paste =)

Reply:
If this script just writes the file, then later we can teach CI to harvest the file, upload it to storage, and include a link to it on the LUCI status page.

Reply:
I filed an proposal for that: https://github.com/flutter/flutter/issues/143409

I'm not going to do it as part of this PR and we can prioritize it against the other stuff we need.  This PR fixes a problem that needs to be addressed asap.